build_test/lib/src/stub_reader.dart

```diff
@@ -22,7 +22,8 @@ class StubAssetReader extends AssetReader implements MultiPackageAssetReader {
       Future.value(null);
 
   @override
-  Stream<AssetId> findAssets(Glob glob, {String package}) => null;
+  Stream<AssetId> findAssets(Glob glob, {String? package}) =>
```

Comment:
I am confused how the other methods don't need to be updated?

Reply:
They all return `Future.value(null)` which apparently isn't a static error, but would fail at runtime.

`StubAssetReader`/`StubAssetWriter` seem to be exclusively used in other tests of the build system, for instance to ensure that a `BuildStepImpl` doesn't allow certain reads or writes. I wasn't sure how migrate the reader: Should the methods throw an `UnsupportedError('stub')`? Some bogus value?

Reply:
Ya we would have to look more at how the tests are used.... this type of pattern is definitely more challenging with null safety.